auth/auth-server-api/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+//! API types for the auth server
+
+#![deny(missing_docs)]
+#![deny(clippy::missing_docs_in_private_items)]
+#![deny(unsafe_code)]
+#![deny(clippy::needless_pass_by_ref_mut)]
+#![feature(trivial_bounds)]
+
+use serde::Deserialize;
+use uuid::Uuid;
+
+// ----------------------
+// | API Key Management |
+// ----------------------
+
+/// The path to create a new API key
+///
+/// POST /api-keys
+pub const API_KEYS_PATH: &str = "api-keys";
+/// The path to mark an API key as inactive
+///
+/// POST /api-keys/{id}/deactivate
+pub const DEACTIVATE_API_KEY_PATH: &str = "deactivate";
+
+/// A request to create a new API key
+#[derive(Debug, Deserialize)]
+pub struct CreateApiKeyRequest {
+    /// The API key id
+    pub id: Uuid,
+    /// The API key secret
+    ///
+    /// Expected as a base64 encoded string
+    pub secret: String,
+    /// The name of the API key
+    pub name: String,
```

Comment:
nit: can we either rename this to `username`, or rename the column on the `ApiKey` table to `name`, so that we refer to these fields consistently when interfacing w/ the API and the DB?

Reply:
it's not quite a username, description is most appropriate, i can change it to that